internal/services/apimanagement/api_management_policy_fragment_resource_test.go

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package apimanagement_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/apimanagement/2022-08-01/policyfragment"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type ApiManagementPolicyFragmentResource struct{}
+
+func TestAccApiManagementPolicyFragment_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_api_management_policy_fragment", "test")
+	r := ApiManagementPolicyFragmentResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("description").HasValue("Some descriptive text"),
+				check.That(data.ResourceName).Key("format").HasValue("xml"),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccApiManagementPolicyFragment_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_api_management_policy_fragment", "test")
+	r := ApiManagementPolicyFragmentResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccApiManagementPolicyFragment_updateDescription(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_api_management_policy_fragment", "test")
+	r := ApiManagementPolicyFragmentResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("description").HasValue("Some descriptive text"),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("description").HasValue("Some descriptive text which is updated"),
```

Comment:
Instead we should add an additional check here for `format`
```suggestion
				check.That(data.ResourceName).Key("description").HasValue("Some descriptive text which is updated"),
				check.That(data.ResourceName).Key("format").HasValue("rawxml"),
```